scalding-core/src/main/scala/com/twitter/scalding/TypedPipeChecker.scala

```diff
@@ -0,0 +1,32 @@
+package com.twitter.scalding
+
+/**
+ * This class is used to assist with testing a TypedPipe
+ */
+object TypedPipeChecker {
+
+  /*
+   * Takes a List and a transform function.
+   * The resulting TypedPipe form the transform will be run through asserts
+   */
+  def checkOutputTransform[T, U, R](input: List[T])(transform: TypedPipe[T] => TypedPipe[U])(assertions: List[U] => R): R =
+    assertions(checkOutputInline(transform(TypedPipe.from(input))))
+
+  /*
+   * Execute a TypedPipe in memory, convert the resulting Iterator to
+   * a list and run it through a function that makes arbitrary
+   * assertions on it.
+   */
+  def checkOutput[T, R](output: TypedPipe[T])(assertions: List[T] => R): R =
```

Comment:
Mentioned above, I don't mind this name too much since it indicates that it should probably only be used for tests.  With mapAll we should also warn that it pulls everything into memory.  mapAllAsList?  If we can imagine people using this function in normal jobs it may be worthwhile to think about other names.